src/main/java/com/blackducksoftware/integration/hub/rest/RestConnection.java

```diff
@@ -0,0 +1,637 @@
+	    }
+	}
+	if (noProxyHosts != null && !noProxyHosts.isEmpty()) {
+	    String noProxyHostsString = null;
+	    for (final Pattern pattern : noProxyHosts) {
+		if (noProxyHostsString == null) {
+		    noProxyHostsString = pattern.toString();
+		} else {
+		    noProxyHostsString = noProxyHostsString + "|"
+			    + pattern.toString();
+		}
+	    }
+	    if (!StringUtils.isBlank(noProxyHostsString)) {
+		System.setProperty("http.nonProxyHosts", noProxyHostsString);
+	    }
+	}
+    }
+
+    /**
+     * Clears the previously set System properties I.E. https.proxyHost,
+     * https.proxyPort, http.proxyHost, http.proxyPort, http.nonProxyHosts
+     *
+     */
+    private void cleanUpOldProxySettings() {
+	System.clearProperty("https.proxyHost");
+	System.clearProperty("https.proxyPort");
+	System.clearProperty("http.proxyHost");
+	System.clearProperty("http.proxyPort");
+	System.clearProperty("http.nonProxyHosts");
+
+	attemptResetProxyCache();
+
+	Authenticator.setDefault(null);
+    }
+
+    private void attemptResetProxyCache() {
+	try {
+	    Class<?> sunAuthCacheValue;
+	    Class<?> sunAuthCache;
+	    Class<?> sunAuthCacheImpl;
+	    try {
+		sunAuthCacheValue = Class
+			.forName("sun.net.www.protocol.http.AuthCacheValue");
+		sunAuthCache = Class
+			.forName("sun.net.www.protocol.http.AuthCache");
+		sunAuthCacheImpl = Class
+			.forName("sun.net.www.protocol.http.AuthCacheImpl");
+	    } catch (final Exception e) {
+		// Must not be using a JDK with sun classes so we abandon this
+		// reset since it is sun specific
+		return;
+	    }
+
+	    final java.lang.reflect.Method m = sunAuthCacheValue
+		    .getDeclaredMethod("setAuthCache", sunAuthCache);
+
+	    final Constructor<?> authCacheImplConstr = sunAuthCacheImpl
+		    .getConstructor();
+	    final Object authCachImp = authCacheImplConstr.newInstance();
+
+	    m.invoke(null, authCachImp);
+
+	} catch (final Exception e) {
+	    logger.error(e.getMessage());
+	}
+    }
+
+    // TODO not sure we want any static methods in this class
+    public static int getResponseStatusCode(ClientResource resource) {
+	return resource.getResponse().getStatus().getCode();
+    }
+
+    /**
+     * This method exists for code symmetry with the other createReference()
+     * method.
+     */
+    // TODO get rid of this
+    static Reference createReference(String url) {
+	return new Reference(url);
+    }
+
+    // TODO get rid of this
+    public static Reference createReference(String baseUrl,
+	    List<String> urlSegments,
+	    Set<AbstractMap.SimpleEntry<String, String>> queryParameters) {
+	Reference queryRef = new Reference(baseUrl);
+	for (String urlSegment : urlSegments) {
+	    queryRef.addSegment(urlSegment);
+	}
+	for (AbstractMap.SimpleEntry<String, String> queryParameter : queryParameters) {
+	    queryRef.addQueryParameter(queryParameter.getKey(),
+		    queryParameter.getValue());
+	}
+	return queryRef;
+    }
+
+    // TODO not sure we want any static methods in this class
```

Comment:
our convention for the most part has been to avoid having static methods unless it is a singleton or possibly a factory.  But we favor creating an instance of the object and using the reference.  If these are utility methods not needed to be a part of RestConnection then creating a utility class that is a singleton would work here.  That's just the convention our team has taken going forward with regards to static methods.